src/app/util/ClientMonitoringRegistrationTable.h

```diff
@@ -19,33 +19,33 @@
 
 #include <app-common/zap-generated/cluster-objects.h>
 #include <lib/core/CHIPConfig.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
 #include <lib/support/CodeUtils.h>
-#include <platform/CHIPDeviceConfig.h>
 
 namespace chip {
 class ClientMonitoringRegistrationTable
 {
 public:
     using MonitoringRegistrationStruct = chip::app::Clusters::ClientMonitoring::Structs::MonitoringRegistration::Type;
 
-    ClientMonitoringRegistrationTable(FabricIndex fabricIndex);
-    ~ClientMonitoringRegistrationTable(){};
+    struct ClientRegistrationEntry : MonitoringRegistrationStruct
+    {
+        bool isValid() { return clientNodeId != 0 && ICid != 0 && fabricIndex != 0; }
+    };
 
-    void SaveToStorage();
+    ClientMonitoringRegistrationTable(PersistentStorageDelegate & storage);
+    ~ClientMonitoringRegistrationTable(){};
 
-    // Getter
-    NodeId getClientNodeId();
-    uint64_t getICid();
-    FabricIndex getFaricIndex();
+    CHIP_ERROR SaveToStorage();
+    CHIP_ERROR LoadFromStorage(FabricIndex fabricIndex);
 
-    // Setter
-    void setClientNodeId(NodeId clientNodeId);
-    void setICid(uint64_t ICid);
-    void setFabricIndex(FabricIndex fabric);
+    ClientRegistrationEntry & getClientRegistrationEntry();
```

Comment:
```suggestion
    ClientRegistrationEntry & GetClientRegistrationEntry();
```

This should have some documentation about how there is only one entry at a time, etc.

Reply:
done